python/tvm/topi/random/kernel.py

```diff
@@ -602,3 +602,57 @@ def normal(gen, mean, scale, out_shape, out_dtype):
     )
 
     return new_gen, random_values
+
+
+def multinomial(gen, probs, num_samples):
+    """Draw samples from a multinomial distribution defined by the input tensor.
+
+    Parameters
+    ----------
+    gen : ThreefryKey
+        Generator state. Can be create with :py:func:`tvm.relay.threefry_key`. This should not be
+        reused in another function, otherwise random numbers will be repeated.
+
+    probs: Tensor[(input_rows, indices), float]
```

Comment:
Traditionally, a multinomial distribution would enforce that these probabilities are all non-negative. It doesn't look like you're enforcing that condition here. Is that intentional? If so, please add a comment describing what should happen in the case of negative probabilities.

Reply:
I dont think there's a good way to check for negative values at runtime without introducing control flow (which we'd like to avoid). Do you think it's a fair interpretation to say that incoming negative values should have 0% chance of being selected? If so, I could insert a `clip` before processing the data to replace negative values with 0.

Reply:
Replacing negative values with zero sounds perfectly reasonable to me, as long as it's documented. Thanks!